wave_matching/include/wave/matching/matcher.hpp

```diff
@@ -0,0 +1,43 @@
+#ifndef __WAVE_MATCHING_MATCHER_HPP__
+#define __WAVE_MATCHING_MATCHER_HPP__
+
+#include <Eigen/Core>
+#include <Eigen/Geometry>
+
+namespace wave {
+namespace matching {
```

Comment:
I'm not sure we need the matching name space, I think we would like to avoid having additional namespaces if we can.